### PR TITLE
Improve error handling and logging

### LINKED ISSUE #5: Top-level logic error handling
The Algolia client is [created and initialized](https://github.com/lukeocodes/netlify-plugin-algolia-index/blob/6145128b67f74229848f55d99cfbdbcc2c5192ec/exporter/index.js#L8) on the top-level scope. If an error was thrown, it would not be properly handled as a user error (using `utils.build.failBuild()`).

Would it be possible to initialize it inside `onPostBuild()` instead, and add some error handling around it?

### PR DESCRIPTION
**Pull request by lukeocodes**
Fixes #5 and #6 with all-round better error handling

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,23 +41,37 @@ module.exports = {
...
-    await exporter(searchIndex)
+    // Export content to Algolia
+    try {
+      const client = algoliasearch(algoliaAppId, algoliaAdminKey)
+      const index = client.initIndex(algoliaIndex)
+      await exporter(index, newIndex)
+    } catch (error) {
+      console.error(error)
+      // Not exporting to search index doesn't fail the entire build
+      build.failPlugin(`Export to Algolia failed - "${error.message}"`)
```
Comment (ehmicky) [inline index.js]:
```suggestion
      build.failPlugin('Export to Algolia failed', { error })
```

Would keep both the error message and the error stack trace.

File: index.js [hunk 2]
```diff
@@ -39,23 +41,37 @@ module.exports = {
...
 
-    // export content to algolia
-    await exporter(searchIndex)
+    // Export content to Algolia
+    try {
+      const client = algoliasearch(algoliaAppId, algoliaAdminKey)
+      const index = client.initIndex(algoliaIndex)
+      await exporter(index, newIndex)
+    } catch (error) {
+      console.error(error)
```
Comment (ehmicky) [inline index.js]: When using `build.fail*()`, the error will already be printed in logs, so I think this `console.error()` might be redundant.

File: index.js (hunk 1 above)
Comment (ehmicky) [inline index.js]: For the two `failPlugin()` in this file: should a Site be deployed to production even if the Algolia index has not been properly built? If not, `failBuild()` should be used instead.

Comment (lukeocodes) [reply index.js]: Yeh I think search missing some records shouldn't prevent new content or changes going live. This was a deliberate move :)

File: index.js (hunk 2 above)
Comment (lukeocodes) [reply index.js]: Ahh good point :)

File: index.js (hunk 1 above)
Comment (lukeocodes) [reply index.js]: I may make this config driven in a later release, actually, to reduce opinionated code

Comment (ehmicky) [reply index.js]: Sounds good!

**Review by ehmicky — APPROVED** [review]

**Pull request merged**